### PR TITLE
Rename UbuntuAdvantage -> UbuntuPro

### PR DESCRIPTION
**Pull request by ogayot**
Nothing too crazy - just replacing UbuntuAdvantage with UbuntuPro wherever it makes sense.

There are still occurrences of "UbuntuAdvantage" for:
* modules & classes that interface with `ubuntu-advantage-tools`.
* integration with cloud-init (which expects the `ubuntu_advantage` key)

We also keep the old "ubuntu-advantage" autoinstall key as a deprecated alias so that we ensure backward-compatibility.

~~Making it a draft so it does not get merged until I open a relevant merge proposal in ubuntu-test-cases.~~
EDIT: https://code.launchpad.net/~ogayot/ubuntu-test-cases/ubuntu-advantage-pro/+merge/416881

ubuntu-pro

Thanks,
Olivier

### CODE REVIEW

File: subiquity/server/controllers/ubuntu_pro.py [hunk 1]
```diff
@@ -34,20 +34,20 @@
...
-class UbuntuAdvantageController(SubiquityController):
-    """ Represent the server-side Ubuntu Advantage controller. """
+class UbuntuProController(SubiquityController):
+    """ Represent the server-side Ubuntu Pro controller. """
 
-    endpoint = API.ubuntu_advantage
+    endpoint = API.ubuntu_pro
 
-    model_name = "ubuntu_advantage"
-    autoinstall_key = "ubuntu-advantage"
```
Comment (dbungert) [inline subiquity/server/controllers/ubuntu_pro.py:50]: We have shipped subiquity where ubuntu-advantage was a valid autoinstall key, and fixing autoinstall errors is kind of hard for end users.
I would like to have a light compatibility layer that is willing to read in something under the ubuntu-advantage name and populates the value under ubuntu-pro.

**Review by dbungert — COMMENTED** [review]
see comment about autoinstall

File: subiquity/server/controllers/ubuntu_pro.py (hunk 1 above)
Comment (ogayot) [reply subiquity/server/controllers/ubuntu_pro.py:50]: Good point, thanks!

**Review by dbungert — APPROVED** [review]

**Comment by ogayot** [comment]
@jpnurmi this PR has API-breaking changes:

HTTP `/ubuntu_advantage`   ->  ` /ubuntu_pro`

**Pull request merged**